src/test/kotlin/MockRasterEndpoint.kt

```diff
@@ -0,0 +1,23 @@
+import io.ktor.client.engine.mock.MockEngine
+import io.ktor.client.engine.mock.MockHttpResponse
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.Url
+import io.ktor.http.fullPath
+import io.ktor.http.hostWithPort
+import kotlinx.coroutines.io.ByteReadChannel
+
+class MockRasterEndpoint(val rasterUrl: String, val fails: Boolean = false) {
+    var hits = 0
+
+    val engine = MockEngine {
+        return@MockEngine if (url.fullUrl.startsWith(rasterUrl) && !fails) {
```

Comment:
What the heck is this?

Reply:
Linking the page we were talking about: https://kotlinlang.org/docs/reference/returns.html

I included the explicit return since the reader might miss the return of the HTTP response. But since it's unnecessary, and the compiler enforces that an HTTP response is returned, it's worth considering removing it.